GVFS/GVFS.Platform.Mac/MacFileSystemVirtualizer.cs

```diff
@@ -80,15 +81,44 @@ public override void Stop()
             string sha)
         {
             // TODO(Mac): Add functional tests that validate file mode is set correctly
-            ushort fileMode = this.FileSystemCallbacks.GitIndexProjection.GetFilePathMode(relativePath);
-            Result result = this.virtualizationInstance.WritePlaceholderFile(
-                relativePath,
-                PlaceholderVersionId,
-                ToVersionIdByteArray(FileSystemVirtualizer.ConvertShaToContentId(sha)),
-                (ulong)endOfFile,
-                fileMode);
+            ushort fileTypeAndMode = this.FileSystemCallbacks.GitIndexProjection.GetFileTypeAndMode(relativePath);
+            GitIndexProjection.FileType fileType = (GitIndexProjection.FileType)(fileTypeAndMode & GitIndexProjection.FileTypeMask);
 
-            return new FileSystemResult(ResultToFSResult(result), unchecked((int)result));
+            if (fileType == GitIndexProjection.FileType.Regular)
+            {
+                Result result = this.virtualizationInstance.WritePlaceholderFile(
+                    relativePath,
+                    PlaceholderVersionId,
+                    ToVersionIdByteArray(FileSystemVirtualizer.ConvertShaToContentId(sha)),
+                    (ulong)endOfFile,
+                    (ushort)(fileTypeAndMode & GitIndexProjection.FileModeMask));
+
+                return new FileSystemResult(ResultToFSResult(result), unchecked((int)result));
+            }
+            else if (fileType == GitIndexProjection.FileType.SymLink)
+            {
+                string symLinkContents;

```

Comment:
What is `symLinkContents`? At first I thought it is trying to get the contents of the target file... but now I think it's maybe just the path of the target file? It would be nice to clarify that.